src/uart/registers.rs

```diff
@@ -97,4 +97,84 @@ impl<const PORT_PTR: usize> ControlRegister<PORT_PTR> {
     /// - 0: Error detection is disabled
     /// - 1: Error detection is enabled
     is_bit_frame_error_detection_enabled}
+
+    bit_impl! {17..=16, RW u16,
+    /// # Baud Clock Source
+    /// Select the source for the baud generator (See Table 12-1)
+    /// - 0: Clock 0
+    /// - 1: Clock 1
+    /// - 2: Clock 2
+    /// - 3: Clock 3
+    set_baud_clock_source,
+    /// # Check Baud Clock Source
+    /// Check the current source of the baud generator
+    /// - 0: Clock 0
+    /// - 1: Clock 1
+    /// - 2: Clock 2
+    /// - 3: Clock 3
+    check_baud_clock_source}
+
+    bit_impl! {15, RW,
+    /// # Baud Clock Enable
+    /// Choose if the baud clock is enabled or not
+    /// - 0: Disabled
+    /// - 1: Enabled
+    set_baud_clock_enable,
+    /// # Is Baud Clock Enabled
+    /// Check if the baud clock is enabled
+    /// - 0: Disabled
+    /// - 1: Enabled
+    is_baud_clock_enable}
+
+    bit_impl! {14, WO, // <- FIXME What should this be? Datasheet says RO, but that does not seem right
+    /// # Hardware Flow Control RTS Deassert Condition.
+    /// Discribes the conditions when RTS is deasserted
+    /// - 0: When FIFO level = C_RX_FIFO_DEPTH, RTS is deasserted
+    /// - 1: When FIFO level >= UART_CTRL.rx_thd_val, RTS is deasserted
+    set_hardware_flow_rts_deassert_condition}
+
+    bit_impl! {13, RW,
+    /// # Hardware Flow Control
+    /// Choose if hardware flow control is enabled, or disabled
+    /// - 0: Disabled
+    /// - 1: Enabled
+    set_hardware_flow_control,
+    /// # Check Hardware Flow Control
+    /// Checks if hardware flow control is enabled
+    /// - 0: Disabled
+    /// - 1: Enabled
+    check_hardware_flow_control}
+
+    bit_impl! {12, RW,
+    /// # Number of stop bits
+    /// Set the number of stop bits
+    /// - 0: 1 stop bit
+    /// - 1: 1.5 stop bit for 5 bit mode, 2 bit mode otherwise
+    set_number_of_stop_bits,
+    /// # Check Number Of Stop Bits
+    /// Check the current number of stop bits
+    /// - 0: 1 stop bit
+    /// - 1: 1.5 stop bit for 5 bit mode, 2 bit mode otherwise
+    check_number_of_stop_bits}
+
+    bit_impl! {11..=10, RW u16,
+    /// # Set Character Length
+    /// Set the number of bits in a character in an uart frame.
+    /// - 0: 5 bits
+    /// - 1: 6 bits
+    /// - 2: 7 bits
+    /// - 3: 8 bits
+    set_character_length,
+    /// # Check Character Length
+    /// Checks the current number of bits in a character in a uart frame
+    /// - 0: 5 bits
+    /// - 1: 6 bits
+    /// - 2: 7 bits
+    /// - 3: 8 bits
+    check_character_length}
+
+    bit_impl! {9, RESET, // FIXME This needs to be renamed / changed
+    /// # Activate Receive FIFO Flush
+    /// Write a 1 to flush the receive fifo
```

Comment:
You probably _shouldn't_ say
> Write a 1 to flush the receive fifo

because this function does not take any prams and could cause the user confusion.